From c4e790e5dbcd05d12910a72e2c84ca9fa74aec85 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Mon, 25 Jul 2016 18:11:10 +0200
Subject: [PATCH 10/11] include: ModemManager-version.h and
 ModemManager-names.h not distributed

These files now depend on different configure options (e.g. if optional
interfaces are enabled or not), so we shouldn't be distributing them.
---
 configure.ac        |  3 +++
 include/Makefile.am | 10 +++++++---
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index f788193f..164f4818 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,6 +93,9 @@ dnl-----------------------------------------------------------------------------
 dnl Build dependencies
 dnl
 
+AC_PATH_PROG(XSLTPROC, xsltproc, AC_MSG_ERROR([xsltproc not found]))
+AC_SUBST(XSLTPROC)
+
 GLIB_MIN_VERSION=2.36.0
 GLIB_BUILD_SYMBOLS="-DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_36 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_36 -DGLIB_DISABLE_DEPRECATION_WARNINGS"
 
diff --git a/include/Makefile.am b/include/Makefile.am
index 37e07dec..3025b53f 100644
--- a/include/Makefile.am
+++ b/include/Makefile.am
@@ -5,12 +5,16 @@ XMLS = $(wildcard $(top_srcdir)/introspection/*.xml)
 includedir = @includedir@/ModemManager
 
 include_HEADERS = \
-	ModemManager-names.h \
 	ModemManager-enums.h \
 	ModemManager-errors.h \
 	ModemManager-compat.h \
+	ModemManager.h \
+	$(NULL)
+
+nodist_include_HEADERS = \
+	ModemManager-names.h \
 	ModemManager-version.h \
-	ModemManager.h
+	$(NULL)
 
 if WITH_INTERFACE_3GPP_USSD
 include_HEADERS += ModemManager-enums-3gpp-ussd.h
@@ -37,7 +41,7 @@ include_HEADERS += ModemManager-enums-firmware.h
 endif
 
 ModemManager-names.h: $(XMLS) $(top_srcdir)/build-aux/header-generator.xsl
-	$(AM_V_GEN) $(XSLTPROC) $(top_srcdir)/build-aux/header-generator.xsl $(top_builddir)/introspection/all.xml > $@
+	$(AM_V_GEN) $(XSLTPROC) --path $(top_srcdir)/introspection $(top_srcdir)/build-aux/header-generator.xsl $(top_builddir)/introspection/all.xml > $@
 
 CLEANFILES = \
 	ModemManager-names.h
-- 
2.14.1

